backend/services/event.py

```diff
@@ -0,0 +1,165 @@
+from fastapi import Depends
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+
+from backend.models.user import User
+from ..database import db_session
+from backend.models.event import Event
+from backend.models.event_details import EventDetails
+from ..entities import EventEntity
+from .permission import PermissionService
+
+class EventNotFoundException(Exception):
+    """EventNotFoundException is raised when trying to access an event that does not exist."""
+
+    def __init__(self, id: str):
+        super().__init__(
+            f'No event found with matching ID: {id}')
+
+class EventService:
+    """Service that performs all of the actions on the `Event` table"""
+
+    # Current SQLAlchemy Session
+    _session: Session
+
+    def __init__(self, session: Session = Depends(db_session), permission: PermissionService = Depends()):
+        """Initializes the `EventService` session"""
+        self._session = session
+        self._permission = permission
+
+    def all(self) -> list[EventDetails]:

```

Comment:
This will ultimately need to be more nuanced, right -- like a guarded by a date range or some other form of pagination? Thinking about what the use case is for this method five years from now, with an assumption that we will not delete old events.